### ISSUE TITLE
Text input dialog

### ISSUE THREAD

**Issue by coolcoder613eb** [beeware/toga#1840]
### What is the problem or limitation you are having?

I am porting a tkinter app to toga,
and had been using `tkinter.simpledialog.askstring`
to prompt for input,
and i could not find any equivalent in toga.

### Describe the solution you'd like

A dialog with a TextInput or a way to make custom dialogs.

### Describe alternatives you've considered

subclassing `toga.Window.factory.dialogs.ConfirmDialog`,
putting the `TextInput` and ok/cancel buttons in the main window

**Comment by freakboy3742** [comment]
Thanks for the suggestion. As a general matter of UI style, Modal dialogs asking for input are generally discouraged in modern UI designs - but there are times when they are the appropriate solution, so it's probably worth adding a "simple dialog" such as TkInter provides.

**Comment by coolcoder613eb** [comment]
will `2` it work on mobile platforms?

**Comment by freakboy3742** [comment]
No, because mobile platforms don't have the concept of multiple windows.

**Comment by coolcoder613eb** [comment]
then can i have the widgets appear in the main window, and the function not return unless the ok or cancel button has been pressed?

**Comment by freakboy3742** [comment]
Sure - you can replace the content of the main window with alternate content, and have those replacements managed by user interaction. Again, the Window example demonstrates this.

**Comment by coolcoder613eb** [comment]
but how do i wait until a button is pressed, without freezing the gui?

**Comment by freakboy3742** [comment]
You don't "wait" - you add a handler to the button, and when the button is pressed, you do whatever needs to be done. Like most GUI frameworks, you need to embrace event-based programming - You don't write your app as a "while True: do app logic" loop; instead, you respond to specific user interactions by adding handlers to buttons and other input widgets.
